--- st.orig/config.def.h
+++ st/config.def.h
@@ -210,9 +210,6 @@ static MouseShortcut mshortcuts[] = {
 static Shortcut shortcuts[] = {
 	/* mask                 keysym          function        argument */
 	{ XK_ANY_MOD,           XK_Break,       sendbreak,      {.i =  0} },
-	{ ControlMask,          XK_Print,       toggleprinter,  {.i =  0} },
-	{ ShiftMask,            XK_Print,       printscreen,    {.i =  0} },
-	{ XK_ANY_MOD,           XK_Print,       printsel,       {.i =  0} },
 	{ TERMMOD,              XK_braceright,  incalpha,       {.f = +.05} },
 	{ TERMMOD,              XK_braceleft,   incalpha,       {.f = -.05} },
 	{ TERMMOD,              XK_quotedbl,    incalpha,       {.f =  0} },
--- st.orig/st.c
+++ st/st.c
@@ -1850,25 +1850,6 @@ csihandle(void)
 		DEFAULT(csiescseq.arg[0], 1);
 		tmoveto(term.c.x, term.c.y+csiescseq.arg[0]);
 		break;
-	case 'i': /* MC -- Media Copy */
-		switch (csiescseq.arg[0]) {
-		case 0:
-			tdump();
-			break;
-		case 1:
-			tdumpline(term.c.y);
-			break;
-		case 2:
-			tdumpsel();
-			break;
-		case 4:
-			term.mode &= ~MODE_PRINT;
-			break;
-		case 5:
-			term.mode |= MODE_PRINT;
-			break;
-		}
-		break;
 	case 'c': /* DA -- Device Attributes */
 		if (csiescseq.arg[0] == 0)
 			ttywrite(vtiden, strlen(vtiden), 0);
@@ -2607,9 +2588,6 @@ tputc(Rune u)
 			width = 1;
 	}
 
-	if (IS_SET(MODE_PRINT))
-		tprinter(c, len);
-
 	/*
 	 * STR sequence must be checked before anything else
 	 * because it uses all following characters until it
