--- st.orig/x.c
+++ st/x.c
@@ -260,6 +260,9 @@ static char *opt_title = NULL;
 
 static uint buttons; /* bit field of pressed buttons */
 
+static Cursor cursor;
+static XColor xmousefg, xmousebg;
+
 void
 clipcopy(const Arg *dummy)
 {
@@ -1250,10 +1253,8 @@ void
 xinit(int cols, int rows)
 {
 	XGCValues gcvalues;
-	Cursor cursor;
 	Window parent, root;
 	pid_t thispid = getpid();
-	XColor xmousefg, xmousebg;
 	XWindowAttributes attr;
 	XVisualInfo vis;
 
@@ -1920,6 +1921,12 @@ xsetmode(int set, unsigned int flags)
 {
 	int mode = win.mode;
 	MODBIT(win.mode, set, flags);
+	if (flags & MODE_MOUSE) {
+		if (win.mode & MODE_MOUSE)
+			XUndefineCursor(xw.dpy, xw.win);
+		else
+			XDefineCursor(xw.dpy, xw.win, cursor);
+	}
 	if ((win.mode & MODE_REVERSE) != (mode & MODE_REVERSE))
 		redraw();
 }
